arviz/plots/kdeplot.py

```diff
@@ -146,3 +168,72 @@ def fast_kde(x, cumulative=False, bw=4.5):
         density = cs_density / cs_density[-1]
 
     return density, xmin, xmax
+
+
+def fast_kde_2d(x, y, circular=False):
+    """
+    2D fft-based Gaussian kernel density estimate (KDE).
+
+    The code was adapted from https://github.com/mfouesneau/faststats
+
+    Parameters
+    ----------
+    x : Numpy array or list
+    y : Numpy array or list
+    circular: bool
+        If True, use circular boundaries. Defaults to False
+    Returns
+    -------
+    grid: A gridded 2D KDE of the input points (x, y)
+    xmin: minimum value of x
+    xmax: maximum value of x
+    ymin: minimum value of y
+    ymax: maximum value of y
+    """
+    x = np.asarray(x, dtype=float)
+    x = x[np.isfinite(x)]
+    y = np.asarray(y, dtype=float)
+    y = y[np.isfinite(y)]
+
+    xmin, xmax = x.min(), x.max()
+    ymin, ymax = y.min(), y.max()
+
+    len_x = len(x)
+    weights = np.ones(len_x)
+    n_x, n_y = [128, 128]  # [256, 256]
+
+    d_x = (xmax - xmin) / (n_x - 1)
+    d_y = (ymax - ymin) / (n_y - 1)
+
+    xyi = np.vstack((x, y)).T
+    xyi -= [xmin, ymin]
+    xyi /= [d_x, d_y]
+    xyi = np.floor(xyi, xyi).T
+
+    scotts_factor = len_x ** (-1 / 6)
+    cov = np.cov(xyi)
+    std_devs = np.diag(cov ** 0.5)
+    kern_nx, kern_ny = np.round(scotts_factor * 2 * np.pi * std_devs)
+
+    inv_cov = np.linalg.inv(cov * scotts_factor ** 2)
+
+    x_x = np.arange(kern_nx) - kern_nx / 2
+    y_y = np.arange(kern_ny) - kern_ny / 2
+    x_x, y_y = np.meshgrid(x_x, y_y)
+
+    kernel = np.vstack((x_x.flatten(), y_y.flatten()))
+    kernel = (inv_cov @ kernel) * kernel
```

Comment:
If we use the @ symbol for matrix multiply I don't think arviz will work with 3.4.
https://docs.python.org/3/whatsnew/3.5.html

Just to double check @ahartikainen  you mentioned 3.4 compatibility is something you wanted?

Reply:
Yes, at least for the first pip release.

Reply:
good catch! Fixed